plugins/kubernetes-backend/src/service/KubernetesProxy.test.ts

```diff
@@ -657,29 +637,135 @@ describe('KubernetesProxy', () => {
         },
       ),
     );
-    permissionApi.authorize.mockResolvedValue([
-      { result: AuthorizeResult.ALLOW },
-    ]);
+
     clusterSupplier.getClusters.mockResolvedValue([
       {
         name: 'cluster1',
         url: 'http://localhost:9999/subpath',
         authProvider: '',
       },
     ]);
+
     authTranslator.decorateClusterDetailsWithAuth.mockImplementation(
       async x => x,
     );
-    const app = express().use(
-      Router().use('/mountpath', proxy.createRequestHandler({ permissionApi })),
-    );
 
-    const requestPromise = request(app)
-      .get('/mountpath/api/v1/namespaces')
-      .set(HEADER_KUBERNETES_CLUSTER, 'cluster1');
-    worker.use(rest.all(requestPromise.url, (req: any) => req.passthrough()));
+    const requestPromise = setupProxyPromise({
+      proxyPath: '/mountpath',
+      requestPath: '/api/v1/namespaces',
+
+      headers: {
+        [HEADER_KUBERNETES_CLUSTER]: 'cluster1',
+      },
+    });
+
     const response = await requestPromise;
 
     expect(response.status).toEqual(200);
   });
+
+  describe('WebSocket', () => {
+    const proxyPath = '/proxy';
+    const proxyPort = 9000;
+    const wsPath = '/ws';
+    const wsPort = 9999;
```

Comment:
Not a huge deal, but using hardcoded ports for these can potentially hurt portability. Probably these ports will be available in most CI environments, but I could imagine them being in use on a dev machine with a bunch of services on localhost.

If I'm not mistaken, in order to make this more robust you can do like [what supertest does](https://github.com/ladjs/supertest/blob/c1b8f9da7b58a5ffe6bf6a1b5056435a21d3c365/lib/test.js#L48-L49) and listen on port "0", which -- on every OS I've used -- will result in the kernel randomly picking an available port for you.

Reply:
I'll have to look at supertest to see how they're getting the port after `listen`, since I will need it for the requests.

Reply:
Could be that this is even necessary for the GitHub workflows to pass -- I think [this error](https://github.com/backstage/backstage/actions/runs/5596220840/jobs/10232998280?pr=18716#step:8:28) has happened multiple times:
> ```
> ● KubernetesProxy › WebSocket › should proxy websocket connections
>
>    FetchError: request to http://localhost:9000/proxy/ws failed, reason: connect ECONNREFUSED ::1:9000
> ```

As for getting the port after calling `listen(0)`, it looks like you can do

```typescript
app.listen(0);
const port = app.address().port;
```

Reply:
Done